lib/zstd.h

```diff
@@ -1096,6 +1102,46 @@ size_t ZSTD_compress_generic_simpleArgs (
                             ZSTD_EndDirective endOp);
 
 
+/** ZSTD_CCtx_params :
+ *
+ *  - ZSTD_createCCtxParams() : Create a ZSTD_CCtx_params structure
+ *  - ZSTD_CCtxParam_setParameter() : Push parameters one by one into an
+ *  existing ZSTD_CCtx_params structure. This is similar to
+ *  ZSTD_CCtx_setParameter().
+ *  - ZSTD_CCtx_applyCCtxParams() : Apply parameters to an existing CCtx. These
+ *  parameters will be applied to all subsequent compression jobs.
+ *  - ZSTD_compress_generic() : Do compression using the CCtx.
+ *  - ZSTD_freeCCtxParams() : Free the memory.
+ *
+ *  This can be used with ZSTD_estimateCCtxSize_opaque() for static allocation. */
+
+ZSTDLIB_API ZSTD_CCtx_params* ZSTD_createCCtxParams(void);
+
+/*! ZSTD_resetCCtxParams() :
+ *  Reset params to default, with the default compression level. */
+ZSTDLIB_API size_t ZSTD_resetCCtxParams(ZSTD_CCtx_params* params);
+
+/*! ZSTD_initCCtxParams() :
+ *  Set the compression and frame parameters of cctxParams according to params.
+ *  All other parameters are reset to their default values. */
+ZSTDLIB_API size_t ZSTD_initCCtxParams(ZSTD_CCtx_params* cctxParams, ZSTD_parameters params);
```

Comment:
Consider also a variant initializing `cctxParams` from `cLevel`. It's probably going to be useful for a user-facing prototype.

Also : if the transformation is guaranteed to succeed, we don't need an error code, so we should probably use this opportunity to return the target type directly.

So it would give something like this : 
```
ZSTD_CCtx_params ZSTD_initCCtxParams(int compressionLevel);
ZSTD_CCtx_params ZSTD_initCCtxParams_advanced(ZSTD_parameters params);
```